### PR TITLE
Milestone 3: Unit-Tests & Validation

### PR DESCRIPTION
**Pull request by SaidRasinlic**
### In this milestone I've successfully completed the following things:

- Added validations in Models
- Created BDD Unit-Tests for all Model's validations and methods

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -3,6 +3,9 @@ class User < ApplicationRecord
   has_many :likes, class_name: 'Like', foreign_key: 'author_id'
   has_many :comments, class_name: 'Comment', foreign_key: 'author_id'
 
+  validates :name, presence: true
+  validates :posts_counter, numericality: { only_integer: true, greater_than_or_equal_to: 0 }
+
   def recent_posts
     posts.last(3)
   end
```
Comment (Gambit142) [inline app/models/user.rb:11]: - [ ] Remember that one of the project requirements is to test all the custom methods in each model. Kindly write a unit test for this method. 👍

**Review by Gambit142 — CHANGES_REQUESTED** [review]
Hi @SaidRasinlic 👋, 

Good job so far!

## Highlights
- All linters checks passed ✔️ 
- Has validations for attributes in each model ✔️ 
- Wrote validation tests and unit tests ✔️ 

However, there are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Kindly check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:

- #### I would advise you to validate the `text` attribute in the `Comment` model because realistically, no user should be able to create a comment without a text. 👍 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

File: app/models/user.rb (hunk 1 above)
Comment (SaidRasinlic) [reply app/models/user.rb:11]: Added additional unit-tests for recent_comments, and increment posts_counter.

**Review by codepantha — APPROVED** [review]
# STATUS: APPROVED 🟢 🍷 

## Hi @SaidRasinlic 👋, 

Well done implementing the required changes 👏 
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 🍷 



### Highlights ⭐ 
✔️ Descriptive PR
✔️ Good commit history
✔️ Correct GitFlow
✔️ No linter errors

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**